dlgr/griduniverse/experiment.py

```diff
@@ -321,36 +312,54 @@ def __init__(self, **kwargs):
             self.color_costs = [2**i for i in range(self.num_colors)]
             random.shuffle(self.color_costs)
 
-        # get food spawning probability distribution function and args
-        self.food_probability_info = {}
+        # get item spawning probability distribution function and args
         self.probability_function_args = []
-        parts = self.food_probability_distribution.split()
+        parts = self.probability_distribution.split()
         if len(parts) > 1:
-            self.food_probability_distribution = parts[0]
+            self.probability_distribution = parts[0]
             self.probability_function_args = parts[1:]
         probability_distribution = "{}_probability_distribution".format(
-            self.food_probability_distribution)
-        self.food_probability_function = getattr(distributions,
-                                                 probability_distribution,
-                                                 None)
-        if self.food_probability_function is None:
+            self.probability_distribution
+        )
+        self.probability_function = getattr(
+            distributions, probability_distribution, None
+        )
+        if self.probability_function is None:
             logger.info(
-                "Unknown food probability distribution: {}.".format(
-                    self.food_probability_distribution))
-            self.food_probability_function = distributions.random_probability_distribution
+                "Unknown item probability distribution: {}.".format(
+                    self.probability_distribution
+                )
+            )
+            self.probability_function = distributions.random_probability_distribution
+
+        # Items and transitions
+        self.item_config = kwargs.get("item_config", DEFAULT_ITEM_CONFIG)
+        self.transition_config = kwargs.get("transition_config", {})
+
+        # public good
+        calories = 0
+        for item_type in self.item_config.values():
+            calories += item_type["calories"]
+        self.public_good = (calories * self.public_good_multiplier) / self.num_players
```

Comment:
Should we be using the _average_ of all the `calories` values here, instead of the sum?

Reply:
I think we need a per item public good value, not a global one. I don't think either the sum or the average across all item type calorie counts makes a lot of sense here.